### ISSUE TITLE
Top-scope variable warning on inline lambda

### ISSUE THREAD

**Issue by steventwheeler** [rodjek/puppet-lint#549]
When running puppet-lint on a class which contains more than an inline lambda a top-scope variable warning is incorrectly issued.
### test.pp

``` puppet
class test {
  $::facts['partitions'].values.map |$data| { $data['mount'] }.each |$mount| {
    notice("Mount: ${mount}")
  }
}
```

```
$ puppet-lint test.pp --no-autoloader_layout-check  --no-documentation-check --with-context
WARNING: top-scope variable being used without an explicit namespace on line 2

  $::facts['partitions'].values.map |$data| { $data['mount'] }.each |$mount| {
                                              ^
```

However, refactoring the code so that the lambda occupies multiple lines does not cause the same warning to be displayed.
### test2.pp

``` puppet
class test2 {
  $::facts['partitions'].values.map |$data| {
    $data['mount']
  }.each |$mount| {
    notice("Mount: ${mount}")
  }
}
```

```
$ puppet-lint test2.pp --no-autoloader_layout-check  --no-documentation-check --with-context
```

Produces no warnings.

```
$ puppet-lint --version
puppet-lint 2.0.2
```

**Comment by cyberious** [comment]
Good to know this isn't just me.

**Comment by dzeleski** [comment]
Same, thought I was going crazy over here. Going back this seems to be an on going issue between versions for a long time. We have just disabled this check for the time being.

```
  $inputs.each |$group_key, $value| {
    $value.each |$setting_key, $setting_value| {
      ini_setting { "/opt/splunk/etc/apps/search/local/inputs.conf_${group_key}_${setting_key}_${setting_value}":
        ensure  => 'present',
        path    => '/opt/splunk/etc/apps/search/local/inputs.conf',
        section => $group_key,
        setting => $setting_key,
        value   => $setting_value,
      }
    }
  }
```

```
Puppet File: \manifests\components\splunk.pp has the following issue(s):
    WARNING: top-scope variable being used without an explicit namespace on line 208
    WARNING: top-scope variable being used without an explicit namespace on line 209
    WARNING: top-scope variable being used without an explicit namespace on line 212
```

First line is 207

**Comment by rnelson0** [comment]
This was resolved with the batch of changes in #636, #637, #638.
